src/wasm-interpreter.h

```diff
@@ -130,6 +130,35 @@ class Flow {
   }
 };
 
+struct FuncData {
```

Comment:
Can this go in `literal.h` alongside `GCData` and `ExnData`? Putting it here creates a weird circular dependency between literal.h and wasm-interpreter.h. Perhaps we can factor out `Instance` as a standalone concept that can be used by both `literal.h` and the interpreter. (FWIW the new interpreter is organized along these lines in anticipation of this use case.)

Reply:
Ok, LGTM for now. In the long run, I still think it would be best if we organize our interpreter and its various components, including values (i.e. `Literal`s), as close to how the spec organizes them as possible.